packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -116,10 +109,23 @@
 			</tera-drilldown-preview>
 		</template>
 	</tera-drilldown>
+	<tera-modal v-if="isNewModelModalVisible">
+		<template #header>
+			<h4>New Model</h4>
+		</template>
+		<form @submit.prevent>
+			<label for="new-model">Enter a unique name for your model</label>
+			<InputText id="new-model" type="text" v-model="newModelName" placeholder="New model" />
+		</form>
+		<template #footer>
+			<Button @click="createModel">Create model</Button>
+			<Button class="p-button-secondary" @click="isNewModelModalVisible = false">Cancel</Button>
+		</template>
+	</tera-modal>
```

Comment:
we can't reuse the `tera-model-modal` here as it strictly creates an empty new model.  We have a unique case here since we already have the model create, we just need to rename it, so custom functions are need.